### PR TITLE
Debian packakes can be installed via URL (for dev releases e.g.)

### PR DESCRIPTION
**Pull request by sirkjohannsen**
This commit enables installation of filebeat via an URL to the deb
package !

I hope it's okay to contribute. I really like your role ! 
I'd even volunteer to write the documentation ;-)

### CODE REVIEW

**Comment by sirkjohannsen** [comment]
@torian - the build fails due to the "block" section. 
Do you maybe want to upgrade to Ansible 2.1 or should I make it compatible with earlier versions ?

**Comment by torian** [comment]
Thanks a lot for your contribution ! Always welcome.

I'd rather keep this compliant with 1.x, at least in the next immediate release, so we give people still using 1.x a chance.
We can then release a new version with the improvements of 2.x

**Comment by leifmadsen** [comment]
@torian I'd say that 2.x is fairly well deployed at this point... I'm not sure there is a reason to avoid 2.x related modules. In fact I'm using 2.1 changes for `yum_repository` stuff. If you're concerned, maybe it'd make sense to version / tag the repo as is, then merge this into master so that 2.x related changes can move forward? It's certainly better than having a repo file, IMHO.

**Comment by sirkjohannsen** [comment]
@torian  I can make it 1.x compatible... just a minute ;-)

**Comment by sirkjohannsen** [comment]
done

**Comment by torian** [comment]
@leifmadsen The idea is not trying to avoid moving forward, but rather provide backward compat for those who are behind (the block directive is a major backward compat breaker).
That being said, with latest changes, we can merge, and tag as `1.0.0`. Next, we can release with the additions of the newest directives.

**Comment by torian** [comment]
@sirkjohannsen Nice one ! If you could update the docs (README and CHANGELOG) that @leifmadsen so generously wrote, we can merge it !

Thanks guys, awesome contributions.

**Comment by sirkjohannsen** [comment]
@torian  - rebased and Readme updated !

**Comment by leifmadsen** [comment]
@torian absolutely, we're definitely on the same page here. Let's get the last set of changes merged, then tag out a 1.x and then add some new hotness to master :)

(I was thinking it might be worth it to have a 1.0 branch maybe, and then that branch can be long live for Ansible 1.9.x, and then master could be for 2.x? Or does this all just seem like too much work, as going forward contributions would need to first land against the 1.0 branch, then merge forward into master... ?)

File: README.md [hunk 1]
```diff
@@ -35,6 +35,15 @@ FileBeat version to use.
 
     filebeat_version: 1.1.1
 
+On Debian based Systems you may use a URL to install a specific .deb
```
Comment (leifmadsen) [inline README.md]: (Probably overridden by comment below.)

s/Systems/systems

Wrap .deb with `.deb`

File: README.md [hunk 2]
```diff
@@ -35,6 +35,15 @@ FileBeat version to use.
...
+To do so set
```
Comment (leifmadsen) [inline README.md]: (Probably overridden by comment below.)

To use a specific `.deb`:

File: README.md [hunk 3]
```diff
@@ -35,6 +35,15 @@ FileBeat version to use.
 
     filebeat_version: 1.1.1
 
+On Debian based Systems you may use a URL to install a specific .deb
+To do so set 
+
+    filebeat_use_apt_repo: false (default true)
```
Comment (leifmadsen) [inline README.md]: The documentation shows the default values, so this should really just be set to the value of `true`. I'd adjust the documentation above to say something like this instead:

> On Debian-based systems, you may use a URL to install a specific `.deb`. To do so, change the following value to `false`, then (optionally) adjust the value of `filebeat_deb_baseurl` (which has a default value set for you).

File: README.md [hunk 4]
```diff
@@ -35,6 +35,15 @@ FileBeat version to use.
...
+
+The following deb base url is set by default:
```
Comment (leifmadsen) [inline README.md]: When `filebeat_use_apt_repo` is set to `false` you can optionally modify the baseurl value to point to another location.

**Comment by leifmadsen** [comment]
@sirkjohannsen other than my primary comment about adjusting some documentation, the implementation looks good to me!

+1

**Comment by sirkjohannsen** [comment]
@leifmadsen  - Modified the README. I hope I got all your points :-)

**Comment by leifmadsen** [comment]
+1!

Only thing I noticed is that you have a typo in your initial commit message. Not sure if @torian cares or not, but the ideal is to rebase and squash everything into a single commit.

**Comment by torian** [comment]
meh, typos happen :P Let's merge and fix it afterwards.

**Comment by leifmadsen** [comment]
+1 w00t

**Comment by torian** [comment]
@sirkjohannsen Seems that using `://` on the apt module is only supported on Ansible 2.1. I've added a couple of tests on vagrant and travis and they are failing.

Could you work on a fix for it ? Most probably a patter like the following would fix it:

```
- name: Download pkg
  ...
  when: not filebeat_use_apt_repo

- name: Install pkg
  ...
  when: not filebeat_use_apt_repo
```

**Comment by sirkjohannsen** [comment]
@torian here you go. 
#6 

Also, what do you think of creating 2.x branch ? I would gladly participate.

**Comment by leifmadsen** [comment]
@sirkjohannsen so there is a 1.x branch now where Ansible 1.x stuff can go. I think at this point master can be opened up for 2.x related changes (right @torian?)

**Comment by torian** [comment]
Absolutely. Will update travis and vagrant tests to reflect the version to be the latest one.

**Pull request merged**